api/types.proto

```diff
@@ -30,12 +30,31 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message Set {
+	message Void {}
+	map<string, Void> val = 1;
+}
+
+message Integer {
+	uint64 val = 1;
+}
+
+message GenericResource {
+	oneof resource {
+		Integer integer = 1;
```

Comment:
Seems like `Set` is unavoidable then. But I still think it would be nice to avoid the extra `Integer` message.